src/main/java/games/strategy/engine/lobby/server/db/UserDao.java

```diff
@@ -4,15 +4,23 @@
 
 public interface UserDao {
 
-  HashedPassword getPassword(String userName);
+  /**
+   * @return null if the user does not exist.
+   */
+  HashedPassword getPassword(String username);
 
-  boolean doesUserExist(String userName);
+  /**
+   * @return null if the user does not exist or the user has no legacy password stored.
+   */
+  HashedPassword getMd5Password(String username);
```

Comment:
In other areas of this PR (as well as the Javadoc above), we use the term "legacy password."  Would it make more sense to rename this method `getLegacyPassword()` instead?